app/models/issue.rb

```diff
@@ -49,4 +51,9 @@ def sub_id
   def self.find_from_project(project, sub_id)
     Issue.find(project.issue_ids[sub_id.to_i - 1])
   end
+
+  def set_sub_id
+    current_id = project.issues.count
```

Comment:
> The sub_id always starts from 1, because it's only set to the number of issues in a project after an issue is created (sub_id = project.issues.count)

Aah, I'm sorry I didn't notice that it was `after_create`, I assumed it'd be `before_save`. 

> bedsides, sub_id will always be unique inside a project (project.issues)

Aah, a bit of a catch there. Under high traffic, it's possible for two issues to end up with the same sub_id. Think of this - you and me try to create a new issue at the same time. If we're both served by different threads, and your issue is created **just** before my `after_create` callback, I'll end up with a `sub_id` that is one more that what it's supposed to be, and you'll end up with the same. 

Normal case: 

```
1. I create my issue
2. I fetch the issues count, update my sub_id
3. You create your issue
4. You fetch the issues count, update your sub_id
```

Failing case: 

```
1. I create my issue
2. You create your issue
3. I fetch the issues count, update my sub_id
4. You fetch the issues count, update your sub_id # Both of us get the same sub_id. 
```

Placing this is a before_save callback with validations in place should greatly reduce the chance of this happening. If we don't want this to happen *at all*, I guess we'll have to set unique index constraints right down in the DB.